frontend/app/routes/public/apply/$id/terms-and-conditions.tsx

```diff
@@ -163,6 +249,28 @@ export default function ApplyIndex({ loaderData, params }: Route.ComponentProps)
           </ButtonLink>
         </div>
       </fetcher.Form>
+      <Dialog open={showDialog} onOpenChange={setShowDialog}>
+        <DialogContent>
+          <DialogHeader>
+            <DialogTitle>{t('apply:terms-and-conditions.dialog.title')}</DialogTitle>
+          </DialogHeader>
+          <DialogDescription>{t('apply:terms-and-conditions.dialog.description')}</DialogDescription>
+          <DialogFooter>
+            <DialogClose asChild>
+              <Button id="confirm-modal-back" disabled={isSubmitting} startIcon={faChevronLeft} variant="default" size="sm">
+                {t('apply:terms-and-conditions.dialog.back-btn')}
+              </Button>
+            </DialogClose>
+            <fetcher.Form method="post" noValidate>
+              <CsrfTokenInput />
+              <input type="hidden" name="doNotConsent" value={CheckboxValue.Yes} />
+              <Button id="exit-application" variant="primary" size="sm" type="submit" disabled={isSubmitting} data-gc-analytics-customclick="ESDC-EDSC:CDCP Online Application Form:Exit application - Terms and Conditions click">
```

Comment:
Just to distinguish this action was performed with the modal to follow existing conventions.

```suggestion
              <Button id="exit-application" variant="primary" size="sm" type="submit" disabled={isSubmitting} data-gc-analytics-customclick="ESDC-EDSC:CDCP Online Application Form:Modal Exit application - Terms and Conditions click">
```